src/memgraph.cpp

```diff
@@ -1204,7 +1205,8 @@ int main(int argc, char **argv) {
                             []() -> nlohmann::json { return query::plan::CallProcedure::GetAndResetCounters(); });
   }
 
-  communication::websocket::Server websocket_server{{"0.0.0.0", 7444}};
+  communication::websocket::SafeAuth safe_auth{&auth};
+  communication::websocket::Server websocket_server{{"0.0.0.0", 7444}, safe_auth};
```

Comment:
`safe_auth` is `copyable` and very cheap i.e., a copy of the ptr, therefore you don't really need to have a member reference. Ideally this will look like:

```
  communication::websocket::Server websocket_server{{"0.0.0.0", 7444}, {&auth}};
```

Reply:
I don't think that SafeAuth is saved to copy since it holds a pointer to auth, which has to be synced across all modules that want to use auth (pointer to lock on `auth` => https://github.com/memgraph/memgraph/blob/00cf230b427f639247d6407a20f6285a3318bd8f/src/communication/websocket/auth.hpp#L29). I guess to signify that it makes sense to make SageAuth uncopiable, what do you think @kostasrim ?